packages/cli-scaffold-graphql-service/index.ts

```diff
@@ -0,0 +1,61 @@
+const yaml = require("js-yaml");
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+const ncp = util.promisify(require("ncp").ncp);
+
+module.exports = [
+    {
+        name: "scaffold-template-apollo-service",
+        type: "scaffold-template",
+        scaffold: {
+            name: "GraphQL Apollo Service",
+            description: "Creates an /api template and fills it in serverless.yml",
+            questions: [
+                {
+                    name: "serviceName",
+                    message: "The name of your service"
+                }
+            ],
+            generate: async ({ input, context }) => {
+                try {
+                    // First we update serverless.yml
+                    const { serviceName, appName } = input;
+
+                    const serverlessJson = yaml.safeLoad(fs.readFileSync(context.apiYaml));
+                    if (serverlessJson[serviceName])
+                        throw new Error(
+                            "This service already exists in serverless.yml! Please pick a different name for it."
+                        );
+
+                    serverlessJson[serviceName] = {
+                        component: "@webiny/serverless-apollo-service",
+                        inputs: {
+                            description: `Apollo service '${serviceName}' scaffoled by the CLI.`,
+                            region: "${vars.region}",
```

Comment:
What if `vars.x`, in this case, `vars.region` does not exist?

When the template is deployed, errors will be thrown.

I guess the scaffold could try to use these variables, but if it determines they don't exist, then we could use sensible defaults?

For example: 

- region: `us-east-1`
- debug: `false`
- vars.apollo: 
```
    server:
      introspection: true
      playground: true
```

--- 

The second idea I had is - ask users via inquirer. I think that might be also a good idea, and especially if we prefill these values in the answers, so users can just hit enter.

We could allow users to type in ${vars.region}, and we could detect if he's referencing a variable here. If so, check if it exists immediately in the inquirer, and show him an error message if referencing was invalid. I would use these ${vars.xyz} values as defaults for every question.

If you ask me, I would go with the second solution.

@Pavel910 @Fsalker

Reply:
`vars.region` and `vars.debug` do already exist in serverless.yml though, don't they?

That's where I got them from 😄

Most other components in there (gateway, security, i18n, ...) use it.

Reply:
Yeah, I understand, but I was thinking it would be a good idea to try to handle this case / remove the possibility of potential errors.

Two more ideas I had in mind...

3) If the vars don't exist in the `serverless.yml`, maybe the scaffold template could print a warning once the scaffolding is done. This way user will know about the missing variables.

4) If vars are not existing, the scaffolding template could just add them at the top of the serverless.yml file, so the user doesn't have to do it by himself. Then, the user can edit these variables. And of course, we can print a message at the end of the scaffolding - telling the user he should edit these to his needs.

All in all, these vars will be present because of other services. But still, we will probably, at one point in time, have to maybe add some other variable, that's not already listed in the sls.yml, and then we'll have to proceed with one of the four solutions (or fifth if you have one in mind).

Now we just need to choose the right solution.

Reply:
Per our conversations, let's go with the solution 4 - add missing variables directly to the serverless.yml, and warn the user about the addition. Also, put sensible defaults where possible.